securedrop_client/storage.py

```diff
@@ -532,12 +521,14 @@ def delete_single_submission_or_reply_on_disk(obj_db: Union[File, Message, Reply
             logging.info('File %s already deleted, skipping', file_to_delete)
 
 
-def rename_file(data_dir: str, filename: str, new_filename: str) -> None:
-    filename, _ = os.path.splitext(filename)
-    new_filename, _ = os.path.splitext(new_filename)
+def rename_file(data_dir: str, file: Union[Message, File, Reply], new_filename: str) -> None:
     try:
-        os.rename(os.path.join(data_dir, filename),
-                  os.path.join(data_dir, new_filename))
+        current_location = file.location(data_dir)
+        new_location = os.path.join(
+            os.path.dirname(current_location),
+            os.path.basename(new_filename)
+        )
+        os.rename(current_location, new_location)
```

Comment:
I believe you can remove this call now, see previous comment about not storing replies and messages on disk

Reply:
OK.